### PR TITLE
Added the code for histogram Equalisation 

### PR DESCRIPTION
**Pull request by ainy-123**
# Description:

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue no.) 
#86 

## Type of change:

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

 
# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

# Screenshots / Video:
input
output

### CODE REVIEW

File: flow2ml/Data_Augumentation.py [hunk 1]
```diff
@@ -109,4 +109,18 @@ def applyShear(self,classPath):
             # saving the image by
             plt.imsave(classPath+"/ShearedImages/Sheared"+image, cv2.cvtColor(Sheared, cv2.COLOR_RGB2BGR))
           except Exception as e:
-            print(f"Shearing operation failed due to {e}")
\ No newline at end of file
+            print(f"Shearing operation failed due to {e}")
+
+  
+  def HistogramEqualisation(image_file):
```
Comment (chebroluharika) [inline flow2ml/Data_Augumentation.py:115]: Take self as one of parameter(just to follow code standards)
And also add documentation in proper format for arguments that you are using

File: flow2ml/Data_Augumentation.py [hunk 2]
```diff
@@ -109,4 +109,18 @@ def applyShear(self,classPath):
...
+    '''class containing method to implement histogram equalization to a given image.'''
+    #read the image and flag is 0 here,to indicate image is loaded in grayscale mode.
+    img=cv2.imread(image_file,0) 
+    #Equalize the Histogram for contrast adjustment.
+    equalised_img=cv2.equalizeHist(img)
+    #stacking both the original and equalised images.
+    result=np.hstack((img,equalised_img))
+    #final image is in output.png file
+    cv2.imwrite("output.png",result)
```
Comment (chebroluharika) [inline flow2ml/Data_Augumentation.py:124]: Follow the directory structure that other augmentation methods are using to save processedImages.

File: flow2ml/Flow.py [hunk 3]
```diff
@@ -125,6 +127,21 @@ def applyAugmentation(self,operations):
       self.update_progress( 100/100.0, f"Augmented all images in {folder}"  )
       progress_i = 0
     print()
+  
+  def deploy_tensorflow_model(model,js=False,lite=False):
```
Comment (chebroluharika) [inline flow2ml/Flow.py:131]: Please remove these changes in this PR as these are not related to this change

File: flow2ml/Data_Augumentation.py [hunk 4]
```diff
@@ -109,4 +109,18 @@ def applyShear(self,classPath):
...
\ No newline at end of file
+            print(f"Shearing operation failed due to {e}")
+
+  
+  def HistogramEqualisation(image_file):
+    '''class containing method to implement histogram equalization to a given image.'''
+    #read the image and flag is 0 here,to indicate image is loaded in grayscale mode.
+    img=cv2.imread(image_file,0) 
+    #Equalize the Histogram for contrast adjustment.
+    equalised_img=cv2.equalizeHist(img)
```
Comment (chebroluharika) [inline flow2ml/Data_Augumentation.py:120]: Add exception handling to this

**Comment by ainy-123** [comment]
@chebroluharika I have done the changes, please check this.

**Comment by chebroluharika** [comment]
Also add this code snippet to call this method in sample code section in Readme

File: flow2ml/Data_Augumentation.py [hunk 5]
```diff
@@ -125,16 +125,21 @@ def HistogramEqualisation(self,classPath):
...
-          gray=cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
-          # applies histogram equalisation to the image.
-          equalised_img=cv2.equalizeHist(gray)
-          HistogramEqualised=np.hstack((img,equalised_img))
-          # saving the image by
-          plt.imsave(classPath+"/HistogramEqualisedImages/HistogramEqualised"+image, cv2.cvtColor(HistogramEqualised, cv2.COLOR_RGB2BGR))
-        except Exception as e:
-          print(f"Histogram Equalisation operation failed due to {e}")
+      if(self.operations['Hist_Equal']==False):
+        raise Exception("Histogram Equalisation should not be done.")
```
Comment (chebroluharika) [inline flow2ml/Data_Augumentation.py:129]: Why do we need to raise Exception here.
Because user may/may not need HE. I think its enough to just add if condition for “true”

Comment (chebroluharika) [reply flow2ml/Data_Augumentation.py:129]: Yes please

**Comment by chebroluharika** [comment]
@ainy-123 take latest from main branch. Then make your changes and make a commit, then you will be able to resolve conflicts

File: flow2ml/Data_Augumentation.py [hunk 6]
```diff
@@ -109,4 +109,35 @@ def applyShear(self,classPath):
...
+            # applies histogram equalisation to the image.
+            equalised_img=cv2.equalizeHist(gray)
+            HistogramEqualised=np.hstack((img,equalised_img))
+            # saving the image by
+            plt.imsave(classPath+"/HistogramEqualisedImages/HistogramEqualised"+image, cv2.cvtColor(HistogramEqualised, cv2.COLOR_RGB2BGR))
+          except Exception as e:
+            print(f"Histogram Equalisation operation failed due to {e}")
+
+      
+
```
Comment (chebroluharika) [inline flow2ml/Data_Augumentation.py:142]: Remove blank lines here and also add screenshots after applying this technique

Comment (ainy-123) [reply flow2ml/Data_Augumentation.py]: > Remove blank lines here and also add screenshots after applying this technique

screenshots added.

**Review by chebroluharika — APPROVED** [review]
